src/ResourceManager/DevTestLabs/Commands.DevTestLabs/Commands/DevTestLabsCmdletBase.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Management.DevTestLabs;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.DevTestLabs
+{
+    public class DevTestLabsCmdletBase : AzureRMCmdlet
+    {
+        internal IDevTestLabsClient _dataServiceClient;
+
+        #region Input Parameter Definitions
+
+        /// <summary>
+        /// Lab name
+        /// </summary>
+        [Parameter(Mandatory = true,
```

Comment:
This makes these parameters mandatory across all parameter sets.  This will prevent, for example, listing across resource groups